client/layout/global-sidebar/main.jsx

```diff
@@ -67,7 +80,7 @@ const GlobalSidebar = ( {
 				<Sidebar className={ className } { ...sidebarProps } onClick={ onClick }>
 					{ requireBackLink && (
 						<div className="sidebar__back-link">
-							<a href="/sites" onClick={ handleBackLinkClick }>
+							<a href={ sidebarBackLinkHref } onClick={ handleBackLinkClick }>
```

Comment:
This is generally working well for me, but I may have found an edge case. If you search for a site like: http://calypso.localhost:3000/sites?search=test , then click your profile link, you now have to click "Back" twice on your profile page to get back to the Sites search results page.

I was able to fix this by forcing the redirect in the `handleBackLinkClick` onClick handler with code along these lines: `page.redirect( event.currentTarget.getAttribute( 'href' ) );`

Reply:
Nice catch!

Reply:
Addressed it via https://github.com/Automattic/wp-calypso/pull/89075/commits/6d031b3198219f3c92f41cc70b567040888470cb.

The reason is when we redirect to the /sites with the `search` query string, the page will update the query string to remove the pagination. However, the `window.location.href` is outdated at this moment and then it redirects back to the previous page.